[Tests] OSError: [Errno 9] Bad file descriptor

Occurred here https://github.com/Tribler/tribler/pull/7121  in https://github.com/Tribler/tribler/actions/runs/3345978272/jobs/5542202695

```python
src/tribler/core/components/libtorrent/restapi/tests/test_downloads_endpoint.py . [ 55%]
...............EE

==================================== ERRORS ====================================
________________ ERROR at setup of test_download_unknown_state _________________

loop = <Looptime_UnixSelectorEventLoop running=False closed=False debug=False>
aiohttp_client = <function aiohttp_client.<locals>.go at 0x7fee72ff30d0>
Exception ignored in: <Finalize object, dead>
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/multiprocessing/util.py", line 224, in __call__
    res = self._callback(*self._args, **self._kwargs)
  File "/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/multiprocessing/util.py", line 464, in close_fds
    os.close(fd)
OSError: [Errno 9] Bad file descriptor
mock_dlmgr = <MagicMock id='140661746441904'>
metadata_store = <tribler.core.components.metadata_store.db.store.MetadataStore object at 0x7fee730103a0>

    @pytest.fixture
    def rest_api(loop, aiohttp_client, mock_dlmgr, metadata_store):  # pylint: disable=unused-argument
    
        endpoint = DownloadsEndpoint(mock_dlmgr, metadata_store=metadata_store)
    
        app = Application(middlewares=[error_middleware])
        app.add_subapp('/downloads', endpoint.app)
>       return loop.run_until_complete(aiohttp_client(app))

src/tribler/core/components/libtorrent/restapi/tests/test_downloads_endpoint.py:31: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/asyncio/base_events.py:603: in run_until_complete
    self.run_forever()
/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/asyncio/base_events.py:570: in run_forever
    self._run_once()
/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/asyncio/base_events.py:[182](https://github.com/Tribler/tribler/actions/runs/3345978272/jobs/5542202695#step:5:183)3: in _run_once
    event_list = self._selector.select(timeout)
../../../.virtualenvs/.venv/lib/python3.8/site-packages/looptime/loops.py:110: in __replaced_select
    ready: list[tuple[Any, Any]] = self.__original_select(timeout=0)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <selectors.EpollSelector object at 0x7fee73027[190](https://github.com/Tribler/tribler/actions/runs/3345978272/jobs/5542202695#step:5:191)>, timeout = 0

    def select(self, timeout=None):
        if timeout is None:
            timeout = -1
        elif timeout <= 0:
            timeout = 0
        else:
            # epoll_wait() has a resolution of 1 millisecond, round away
            # from zero to wait *at least* timeout seconds.
            timeout = math.ceil(timeout * 1e3) * 1e-3
    
        # epoll_wait() expects `maxevents` to be greater than zero;
        # we want to make sure that `select()` can be called when no
        # FD is registered.
        max_ev = max(len(self._fd_to_key), 1)
    
        ready = []
        try:
>           fd_event_list = self._selector.poll(timeout, max_ev)
E           OSError: [Errno 9] Bad file descriptor

/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/selectors.py:468: OSError
_______________ ERROR at teardown of test_download_unknown_state _______________

    def finalizer() -> None:
        """Yield again, to finalize."""
    
        async def async_finalizer() -> None:
            try:
                await gen_obj.__anext__()
            except StopAsyncIteration:
                pass
            else:
                msg = "Async generator fixture didn't stop."
                msg += "Yield only once."
                raise ValueError(msg)
    
>       event_loop.run_until_complete(async_finalizer())

../../../.virtualenvs/.venv/lib/python3.8/site-packages/pytest_asyncio/plugin.py:291:
```

https://github.com/Tribler/tribler/actions/runs/3363436019/jobs/5576574607 in https://github.com/Tribler/tribler/pull/7129

```python
==================================== ERRORS ====================================
__________________ ERROR at teardown of test_stream_download ___________________

    def finalizer() -> None:
        """Yield again, to finalize."""
    
        async def async_finalizer() -> None:
            try:
                await gen_obj.__anext__()
            except StopAsyncIteration:
                pass
            else:
                msg = "Async generator fixture didn't stop."
                msg += "Yield only once."
                raise ValueError(msg)
    
>       event_loop.run_until_complete(async_finalizer())

../../../.virtualenvs/.venv/lib/python3.8/site-packages/pytest_asyncio/plugin.py:291: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/asyncio/base_events.py:616: in run_until_complete
    return future.result()
../../../.virtualenvs/.venv/lib/python3.8/site-packages/pytest_asyncio/plugin.py:283: in async_finalizer
    await gen_obj.__anext__()
../../../.virtualenvs/.venv/lib/python3.8/site-packages/pytest_aiohttp/plugin.py:172: in aiohttp_client
    await clients.pop().close()
../../../.virtualenvs/.venv/lib/python3.8/site-packages/aiohttp/test_utils.py:398: in close
    await self._server.close()
../../../.virtualenvs/.venv/lib/python3.8/site-packages/aiohttp/test_utils.py:194: in close
    await self.runner.cleanup()
../../../.virtualenvs/.venv/lib/python3.8/site-packages/aiohttp/web_runner.py:293: in cleanup
    await site.stop()
../../../.virtualenvs/.venv/lib/python3.8/site-packages/aiohttp/web_runner.py:72: in stop
    self._server.close()
/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/asyncio/base_events.py:337: in close
    self._loop._stop_serving(sock)
/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/asyncio/selector_events.py:602: in _stop_serving
    sock.close()
/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/socket.py:500: in close
    self._real_close()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <socket.socket [closed] fd=-1, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0>
_ss = <class '_socket.socket'>

    def _real_close(self, _ss=_socket.socket):
        # This function should not reference any globals. See issue #808[164](https://github.com/Tribler/tribler/actions/runs/3363436019/jobs/5576574607#step:4:165).
>       _ss.close(self)
E       OSError: [Errno 9] Bad file descriptor

/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/socket.py:494: OSError
----------------------------- Captured stderr call -----------------------------
```

https://github.com/Tribler/tribler/actions/runs/3367858797/jobs/5585760856 in https://github.com/Tribler/tribler/pull/7112

```python
src/tribler/core/components/gigachannel/community/tests/test_gigachannel_community.py . [ 18%]
.F
Exception ignored in: <Finalize object, dead>
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/multiprocessing/util.py", line 224, in __call__

    res = self._callback(*self._args, **self._kwargs)
=================================== FAILURES ===================================
  File "/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/multiprocessing/util.py", line 464, in close_fds
    os.close(fd)
OSError: [Errno 9] Bad file descriptor
--- Logging error ---
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/logging/handlers.py", line 69, in emit
    if self.shouldRollover(record):
  File "/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/logging/handlers.py", line 186, in shouldRollover
    self.stream.seek(0, 2)  #due to non-posix-compliant Windows feature
OSError: [Errno 29] Illegal seek
Call stack:
Logged from file process_checker.py, line [85](https://github.com/Tribler/tribler/actions/runs/3367858797/jobs/5585760856#step:5:86)
Message: 'Get PID from the lock file'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/logging/handlers.py", line 69, in emit
    if self.shouldRollover(record):
  File "/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/logging/handlers.py", line 1[86](https://github.com/Tribler/tribler/actions/runs/3367858797/jobs/5585760856#step:5:87), in shouldRollover
    self.stream.seek(0, 2)  #due to non-posix-compliant Windows feature
OSError: [Errno 29] Illegal seek
Call stack:
Logged from file process_checker.py, line 91
Message: FileNotFoundError(2, 'No such file or directory')
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/logging/handlers.py", line 69, in emit
    if self.shouldRollover(record):
  File "/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/logging/handlers.py", line 186, in shouldRollover
    self.stream.seek(0, 2)  #due to non-posix-compliant Windows feature
OSError: [Errno 29] Illegal seek
Call stack:
Logged from file process_checker.py, line 38
Message: 'Lock file: /tmp/pytest-of-runner/pytest-0/test_restart_tribler0/triblerd.lock'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/logging/handlers.py", line 69, in emit
    if self.shouldRollover(record):
  File "/opt/hostedtoolcache/Python/3.8.14/x64/lib/python3.8/logging/handlers.py", line 186, in shouldRollover
    self.stream.seek(0, 2)  #due to non-posix-compliant Windows feature
OSError: [Errno 29] Illegal seek
Call stack:
Logged from file process_checker.py, line 165
Message: 'Restart Tribler'
Arguments: ()
```

--- a/src/tribler/core/components/knowledge/restapi/tests/test_knowledge_endpoint.py
+++ b/src/tribler/core/components/knowledge/restapi/tests/test_knowledge_endpoint.py
@@ -30,7 +30,8 @@ def knowledge_endpoint(knowledge_db):
 def rest_api(loop, aiohttp_client, knowledge_endpoint):
     app = Application()
     app.add_subapp('/knowledge', knowledge_endpoint.app)
-    return loop.run_until_complete(aiohttp_client(app))
+    yield loop.run_until_complete(aiohttp_client(app))
+    app.shutdown()
 
 
 def tag_to_statement(tag: str) -> Dict:

--- a/src/tribler/core/components/libtorrent/restapi/tests/test_downloads_endpoint.py
+++ b/src/tribler/core/components/libtorrent/restapi/tests/test_downloads_endpoint.py
@@ -28,7 +28,8 @@ def rest_api(loop, aiohttp_client, mock_dlmgr, metadata_store):  # pylint: disab
 
     app = Application(middlewares=[error_middleware])
     app.add_subapp('/downloads', endpoint.app)
-    return loop.run_until_complete(aiohttp_client(app))
+    yield loop.run_until_complete(aiohttp_client(app))
+    app.shutdown()
 
 
 def get_hex_infohash(tdef):

--- a/src/tribler/core/components/libtorrent/restapi/tests/test_libtorrent_endpoint.py
+++ b/src/tribler/core/components/libtorrent/restapi/tests/test_libtorrent_endpoint.py
@@ -19,7 +19,8 @@ def endpoint(mock_dlmgr, mock_lt_session):
 def rest_api(loop, aiohttp_client, endpoint):  # pylint: disable=unused-argument
     app = Application(middlewares=[error_middleware])
     app.add_subapp('/libtorrent', endpoint.app)
-    return loop.run_until_complete(aiohttp_client(app))
+    yield loop.run_until_complete(aiohttp_client(app))
+    app.shutdown()
 
 
 @pytest.fixture

--- a/src/tribler/core/components/libtorrent/restapi/tests/test_torrentinfo_endpoint.py
+++ b/src/tribler/core/components/libtorrent/restapi/tests/test_torrentinfo_endpoint.py
@@ -53,7 +53,8 @@ def endpoint(download_manager):
 def rest_api(loop, aiohttp_client, endpoint):  # pylint: disable=unused-argument
     app = Application(middlewares=[error_middleware])
     app.add_subapp('/torrentinfo', endpoint.app)
-    return loop.run_until_complete(aiohttp_client(app))
+    yield loop.run_until_complete(aiohttp_client(app))
+    app.shutdown()
 
 
 async def test_get_torrentinfo_escaped_characters(tmp_path, rest_api):

--- a/src/tribler/core/components/metadata_store/restapi/tests/test_channels_endpoint.py
+++ b/src/tribler/core/components/metadata_store/restapi/tests/test_channels_endpoint.py
@@ -57,7 +57,8 @@ def rest_api(loop, aiohttp_client, mock_dlmgr, metadata_store, knowledge_db):  #
     app = Application(middlewares=[error_middleware])
     app.add_subapp('/channels', channels_endpoint.app)
     app.add_subapp('/collections', collections_endpoint.app)
-    return loop.run_until_complete(aiohttp_client(app))
+    yield loop.run_until_complete(aiohttp_client(app))
+    app.shutdown()
 
 
 async def test_get_channels(rest_api, add_fake_torrents_channels, add_subscribed_and_not_downloaded_channel, mock_dlmgr,

--- a/src/tribler/core/components/metadata_store/restapi/tests/test_metadata_endpoint.py
+++ b/src/tribler/core/components/metadata_store/restapi/tests/test_metadata_endpoint.py
@@ -48,7 +48,8 @@ def rest_api(loop, aiohttp_client, torrent_checker, metadata_store):  # pylint: 
 
     app = Application(middlewares=[error_middleware])
     app.add_subapp('/metadata', endpoint.app)
-    return loop.run_until_complete(aiohttp_client(app))
+    yield loop.run_until_complete(aiohttp_client(app))
+    app.shutdown()
 
 
 async def test_update_multiple_metadata_entries(metadata_store, add_fake_torrents_channels, rest_api):

--- a/src/tribler/core/components/metadata_store/restapi/tests/test_remote_query_endpoint.py
+++ b/src/tribler/core/components/metadata_store/restapi/tests/test_remote_query_endpoint.py
@@ -34,7 +34,8 @@ def endpoint(mock_gigachannel_community, metadata_store):  # pylint: disable=W06
 def rest_api(loop, aiohttp_client, endpoint):  # pylint: disable=unused-argument
     app = Application(middlewares=[error_middleware])
     app.add_subapp('/remote_query', endpoint.app)
-    return loop.run_until_complete(aiohttp_client(app))
+    yield loop.run_until_complete(aiohttp_client(app))
+    app.shutdown()
 
 
 async def test_create_remote_search_request(rest_api, endpoint, mock_gigachannel_community):  # pylint: disable=W0621

--- a/src/tribler/core/components/metadata_store/restapi/tests/test_search_endpoint.py
+++ b/src/tribler/core/components/metadata_store/restapi/tests/test_search_endpoint.py
@@ -37,7 +37,8 @@ def rest_api(loop, needle_in_haystack_mds, aiohttp_client, knowledge_db):
     channels_endpoint = SearchEndpoint(needle_in_haystack_mds, knowledge_db=knowledge_db)
     app = Application()
     app.add_subapp('/search', channels_endpoint.app)
-    return loop.run_until_complete(aiohttp_client(app))
+    yield loop.run_until_complete(aiohttp_client(app))
+    app.shutdown()
 
 
 async def test_search_wrong_mdtype(rest_api):

--- a/src/tribler/core/components/restapi/rest/tests/test_create_torrent_endpoint.py
+++ b/src/tribler/core/components/restapi/rest/tests/test_create_torrent_endpoint.py
@@ -20,7 +20,8 @@ def endpoint():
 def rest_api(loop, aiohttp_client, endpoint):  # pylint: disable=unused-argument
     app = Application(middlewares=[error_middleware])
     app.add_subapp('/createtorrent', endpoint.app)
-    return loop.run_until_complete(aiohttp_client(app))
+    yield loop.run_until_complete(aiohttp_client(app))
+    app.shutdown()
 
 
 async def test_create_torrent(rest_api, tmp_path, endpoint):

--- a/src/tribler/core/components/restapi/rest/tests/test_debug_endpoint.py
+++ b/src/tribler/core/components/restapi/rest/tests/test_debug_endpoint.py
@@ -41,7 +41,8 @@ def rest_api(loop, aiohttp_client, mock_tunnel_community, endpoint):  # pylint: 
 
     app = Application(middlewares=[error_middleware])
     app.add_subapp('/debug', endpoint.app)
-    return loop.run_until_complete(aiohttp_client(app))
+    yield loop.run_until_complete(aiohttp_client(app))
+    app.shutdown()
 
 
 async def test_get_slots(rest_api, mock_tunnel_community):

--- a/src/tribler/core/components/restapi/rest/tests/test_settings_endpoint.py
+++ b/src/tribler/core/components/restapi/rest/tests/test_settings_endpoint.py
@@ -24,7 +24,8 @@ def endpoint(tribler_config):
 def rest_api(loop, aiohttp_client, endpoint):  # pylint: disable=unused-argument
     app = Application(middlewares=[error_middleware])
     app.add_subapp('/settings', endpoint.app)
-    return loop.run_until_complete(aiohttp_client(app))
+    yield loop.run_until_complete(aiohttp_client(app))
+    app.shutdown()
 
 
 def verify_settings(settings_dict):

--- a/src/tribler/core/components/restapi/rest/tests/test_shutdown_endpoint.py
+++ b/src/tribler/core/components/restapi/rest/tests/test_shutdown_endpoint.py
@@ -15,10 +15,10 @@ def endpoint():
 
 @pytest.fixture
 def rest_api(loop, aiohttp_client, endpoint):  # pylint: disable=unused-argument
-
     app = Application(middlewares=[error_middleware])
     app.add_subapp('/shutdown', endpoint.app)
-    return loop.run_until_complete(aiohttp_client(app))
+    yield loop.run_until_complete(aiohttp_client(app))
+    app.shutdown()
 
 
 async def test_shutdown(rest_api, endpoint):

--- a/src/tribler/core/components/restapi/rest/tests/test_statistics_endpoint.py
+++ b/src/tribler/core/components/restapi/rest/tests/test_statistics_endpoint.py
@@ -33,7 +33,8 @@ def endpoint():
 def rest_api(loop, aiohttp_client, endpoint):  # pylint: disable=unused-argument
     app = Application(middlewares=[error_middleware])
     app.add_subapp('/statistics', endpoint.app)
-    return loop.run_until_complete(aiohttp_client(app))
+    yield loop.run_until_complete(aiohttp_client(app))
+    app.shutdown()
 
 
 async def test_get_tribler_statistics(rest_api, endpoint, metadata_store):

--- a/src/tribler/core/components/restapi/rest/tests/test_trustview_endpoint.py
+++ b/src/tribler/core/components/restapi/rest/tests/test_trustview_endpoint.py
@@ -25,7 +25,8 @@ def endpoint(bandwidth_db):  # pylint: disable=W0621
 def rest_api(loop, aiohttp_client, endpoint):  # pylint: disable=unused-argument
     app = Application(middlewares=[error_middleware])
     app.add_subapp('/trustview', endpoint.app)
-    return loop.run_until_complete(aiohttp_client(app))
+    yield loop.run_until_complete(aiohttp_client(app))
+    app.shutdown()
 
 
 @pytest.fixture

--- a/src/tribler/core/logger/logger.py
+++ b/src/tribler/core/logger/logger.py
@@ -7,7 +7,6 @@ import yaml
 
 LOG_CONFIG_FILENAME = 'logger.yaml'
 
-
 logger = logging.getLogger(__name__)
 
 

